lib/awful/ewmh.lua

```diff
@@ -247,6 +247,10 @@ local context_mapper = {
 function ewmh.geometry(c, context, hints)
     local layout = c.screen.selected_tag and c.screen.selected_tag.layout or nil
 
+    --TODO handle `client_maximize_horizontal` and `client_maximize_vertical`
+    -- so the clients can request being maximized. Currently they can set the
+    -- atom, but it is ignored.
```

Comment:
This just breaks client-induced maximisation and is thus a regression (well, replaces one regression with another regression). Sorry.